packages/state-transition/src/slot/upgradeStateToFulu.ts

```diff
@@ -0,0 +1,28 @@
+import {ssz} from "@lodestar/types";
+import {getCachedBeaconState} from "../cache/stateCache.js";
+import {CachedBeaconStateElectra, CachedBeaconStateFulu} from "../types.js";
+
+/**
+ * Upgrade a state from Capella to Deneb.
+ */
+export function upgradeStateToFulu(stateElectra: CachedBeaconStateElectra): CachedBeaconStateFulu {
+  const {config} = stateElectra;
+
+  const stateElectraNode = ssz.electra.BeaconState.commitViewDU(stateElectra);
+  const stateFuluView = ssz.electra.BeaconState.getViewDU(stateElectraNode);
+
+  const stateFulu = getCachedBeaconState(stateFuluView, stateElectra);
+
+  stateFulu.fork = ssz.phase0.Fork.toViewDU({
+    previousVersion: stateElectra.fork.currentVersion,
+    currentVersion: config.FULU_FORK_VERSION,
+    epoch: stateElectra.epochCtx.epoch,
+  });
+
+  stateFulu.commit();
+  // Clear cache to ensure the cache of capella fields is not used by new deneb fields
+  // biome-ignore lint/complexity/useLiteralKeys: It is a protected attribute
+  stateFulu["clearCache"]();
```

Comment:
Don't think you need to clear cache here. Both peerdas and focil don't modify the structure in the beacon state.